addons/survey/models/survey_user.py

```diff
@@ -370,8 +370,13 @@ def _check_answer_type(self):
                 field_name = 'value_%s' % line.answer_type
             else:
                 field_name = False
-            if field_name and not line[field_name]:
-                raise ValidationError(_('The answer must be in the right type'))
+            if line.answer_type == 'numerical_box':
+                # exceptionally allow 0 as a valid number
+                if line[field_name] is False:
+                  raise ValidationError(_('The answer must be in the right type'))
+            else:
+                if field_name and not line[field_name]:
+                    raise ValidationError(_('The answer must be in the right type'))
```

Comment:
Suggestion:

```py
# we allow 0 for numerical box
if field_name and not line[field_name] and (line.answer_type != 'numerical_box' or line[field_name] != 0):
    raise ValidationError(_('The answer must be in the right type'))
```

Reply:
Agreed your way is much better. Changed it.